### PR TITLE
Enable overriding Heapster resource requirements in GCP

### PR DESCRIPTION
**Pull request by kawych**
This PR enables to override Heapster resource requirements in GCP.

**Release note:**
```release-note
```

### CODE REVIEW

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -302,8 +302,13 @@ fi
 FLUENTD_GCP_MEMORY_LIMIT="${FLUENTD_GCP_MEMORY_LIMIT:-300Mi}"
 FLUENTD_GCP_CPU_REQUEST="${FLUENTD_GCP_CPU_REQUEST:-100m}"
 FLUENTD_GCP_MEMORY_REQUEST="${FLUENTD_GCP_MEMORY_REQUEST:-200Mi}"
+# Heapster requirements
```
Comment (piosz) [inline cluster/gce/config-default.sh]: Could you please add the same entry for config-test.sh?

**Comment by piosz** [comment]
One nit, lgtmed too early ;)

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (gmarek) [inline cluster/gce/config-default.sh]: Add empty line

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -302,8 +302,13 @@ fi
...
+HEAPSTER_GCP_BASE_MEMORY="${HEAPSTER_GCP_BASE_MEMORY:-140Mi}"
+HEAPSTER_GCP_MEMORY_PER_NODE="${HEAPSTER_GCP_MEMORY_PER_NODE:-4}"
+HEAPSTER_GCP_BASE_CPU="${HEAPSTER_GCP_BASE_CPU:-80m}"
+HEAPSTER_GCP_CPU_PER_NODE="${HEAPSTER_GCP_CPU_PER_NODE:-0.5}"
 # Adding to PROVIDER_VARS, since this is GCP-specific.
```
Comment (gmarek) [inline cluster/gce/config-default.sh]: Add empty line

File: cluster/gce/config-test.sh [hunk 3]
```diff
@@ -359,8 +359,12 @@ fi
 FLUENTD_GCP_MEMORY_LIMIT="${FLUENTD_GCP_MEMORY_LIMIT:-300Mi}"
 FLUENTD_GCP_CPU_REQUEST="${FLUENTD_GCP_CPU_REQUEST:-100m}"
 FLUENTD_GCP_MEMORY_REQUEST="${FLUENTD_GCP_MEMORY_REQUEST:-200Mi}"
+HEAPSTER_GCP_BASE_MEMORY="${HEAPSTER_GCP_BASE_MEMORY:-140Mi}"
+HEAPSTER_GCP_MEMORY_PER_NODE="${HEAPSTER_GCP_MEMORY_PER_NODE:-4}"
+HEAPSTER_GCP_BASE_CPU="${HEAPSTER_GCP_BASE_CPU:-80m}"
+HEAPSTER_GCP_CPU_PER_NODE="${HEAPSTER_GCP_CPU_PER_NODE:-0.5}"
 # Adding to PROVIDER_VARS, since this is GCP-specific.
-PROVIDER_VARS="${PROVIDER_VARS:-} FLUENTD_GCP_MEMORY_LIMIT FLUENTD_GCP_CPU_REQUEST FLUENTD_GCP_MEMORY_REQUEST"
+PROVIDER_VARS="${PROVIDER_VARS:-} FLUENTD_GCP_MEMORY_LIMIT FLUENTD_GCP_CPU_REQUEST FLUENTD_GCP_MEMORY_REQUEST HEAPSTER_GCP_BASE_MEMORY HEAPSTER_GCP_MEMORY_PER_NODE HEAPSTER_GCP_BASE_CPU HEAPSTER_GCP_CPU_PER_NODE"
```
Comment (gmarek) [inline cluster/gce/config-test.sh]: empty line

File: cluster/gce/config-default.sh (hunk 2 above)
Comment (kawych) [reply cluster/gce/config-default.sh]: Done.

File: cluster/gce/config-default.sh (hunk 1 above)
Comment (kawych) [reply cluster/gce/config-default.sh]: Done.

File: cluster/gce/config-test.sh (hunk 3 above)
Comment (kawych) [reply cluster/gce/config-test.sh]: Done.

**Comment by piosz** [comment]
I've just realized that it might break GKE (see https://github.com/kubernetes/kubernetes/pull/48425#issuecomment-314130996). Removing lgtm label and let's discuss it f2f.

**Comment by kawych** [comment]
@piosz, I've updated the change as discussed to avoid breaking GKE

**Comment by piosz** [comment]
/lgtm
Thanks!

**Pull request merged**